dpe/src/x509.rs

```diff
@@ -2526,36 +2751,26 @@ fn create_dpe_cert_or_csr(
             if issuer_len > MAX_ISSUER_NAME_SIZE {
                 return Err(DpeErrorCode::InternalError);
             }
-            match pub_key {
-                PubKey::Ecdsa(ref exported_pub_key) => {
-                    let cert_validity = env.platform.get_cert_validity()?;
-                    let mut bytes_written = scratch_writer.encode_ecdsa_tbs(
-                        &subject_name.serial.bytes()[..20], // Serial number must be truncated to 20 bytes
-                        &issuer_name[..issuer_len],
-                        &subject_name,
-                        exported_pub_key,
-                        &measurements,
-                        &cert_validity,
-                    )?;
-                    if bytes_written > MAX_CERT_SIZE {
-                        return Err(DpeErrorCode::InternalError);
-                    }
-                    let tbs_digest = env.crypto.hash(&scratch_buf[..bytes_written])?;
-
-                    // Use this pattern to keep the linter happy + introduce a compiler error when
-                    // ML-DSA is enabled so we remember to update these cases.
-                    let sig = match env.crypto.sign_with_alias(&tbs_digest) {
-                        Ok(Signature::Ecdsa(sig)) => sig,
-                        Err(e) => Err(e)?,
-                    };
-
-                    let mut cert_writer =
-                        CertWriter::new(output_cert_or_csr, args.dice_extensions_are_critical);
-                    bytes_written = cert_writer
-                        .encode_ecdsa_certificate(&scratch_buf[..bytes_written], &sig)?;
-                    u32::try_from(bytes_written).map_err(|_| DpeErrorCode::InternalError)?
-                }
+            let cert_validity = env.platform.get_cert_validity()?;
+            let mut bytes_written = scratch_writer.encode_tbs(
+                &subject_name.serial.bytes()[..20], // Serial number must be truncated to 20 bytes
+                &issuer_name[..issuer_len],
+                &subject_name,
+                &pub_key,
+                &measurements,
+                &cert_validity,
+            )?;
+            if bytes_written > MAX_CERT_SIZE {
+                return Err(DpeErrorCode::InternalError);
             }
+            let tbs_digest = env.crypto.hash(&scratch_buf[..bytes_written])?;
+
+            // TODO(clundin): We need to make sure we handle ExternalMu well here.
+            let sig = env.crypto.sign_with_alias(&tbs_digest)?;
```

Comment:
This part is interesting. Since we have all the data for this signature in SRAM we have two choices:

1. Just pass the whole buffer to the ML-DSA peripheral
2. Compute mu and pass that

(1) is probably slightly more efficient. Although we would need to change `sign_with_alias` to accept the whole message.

If we decide to do (2), we could add a crypto API with something like `prepare_tbs` which would hash for ECDSA and compute mu for ML-DSA.

Reply:
Nice. I like that. We can handle in a follow-up PR